modules/gdscript/language_server/godot_lsp.h

```diff
@@ -916,6 +916,13 @@ static const int PlainText = 1;
 static const int Snippet = 2;
 }; // namespace InsertTextFormat
 
+namespace CompletionItemTag {
+/**
+ * Render a completion as obsolete, usually using a strike-out.
+ */
+static const int Deprecated = 1;
+}; // namespace CompletionItemTag
+
```

Comment:
We don't have client capability tracking, tags are only available from LSP 3.15.0. IMO we should stick with `deprecated`.

Reply:
According to the [official specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem), it sounds like `CompletionItem.deprecated` is itself deprecated:
```typescript
export interface CompletionItem {
	...
	/**
	 * Indicates if this item is deprecated.
	 *
	 * @deprecated Use `tags` instead if supported.
	 */
	deprecated?: boolean;
	...
}
```
I thought I checked and found that VS Code, at least, didn't support `CompletionItem.deprecated`, but I can take another look. 

It looks like we may also be able to check the client's capabilities via [`CompletionClientCapabilities`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities), including whether or not they support either `CompletionItem.deprecated` or `CompletionItemTag.Deprecated`.

Reply:
> It looks like we may also be able to check the client's capabilities via [`CompletionClientCapabilities`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities), including whether or not they support either `CompletionItem.deprecated` or `CompletionItemTag.Deprecated`.

I know, that's how it's supposed to work. But our server implementation doesn't implement capability tracking. We just ignore the capabilities that a client sends. I was hoping `deprecated` would work even if it is deprecated in the protocol. When we add tags without proper capability tracking we risk breaking other client implementations that aren't up to date.

I originally didn't want to implement capability tracking before limiting clients to 1 client per server (see https://github.com/godotengine/godot-proposals/issues/11056). But if this becomes a blocker it should be possible to just save them per client for now (at the risk of breaking some assumptions when multiple clients are connected).

Reply:
Great news on this - looks like VS Code does still support the `deprecated` field, without using the tags, unless there's some building/caching weirdness going on. I commented out all of the instances of `tags` and replaced them with the `deprecated` field, and it seems to still be doing the strikethrough.

For now, I'll remove the tags entirely and just use the `deprecated` field :)